bundled translations: use the locale and add docs (#6849)
slint-ui/slint: docs/src/content/docs/concepts/development/localization.md

For example, if you're using the above and the user's locale is `fr`,
Slint looks for `my_application.mo` in the `lang/fr/LC_MESSAGES/` directory.

## Bundled Translations

Bundled translations embed the translated strings directly into your application binary.
This approach is ideal for platforms like WASM or microcontrollers where `gettext` is unavailable.

Configure the Slint compiler to bundle translations by giving it a path to the translations.
Translation files should be organized as:

```
path/<lang>/LC_MESSAGES/<domain>.po
```

### Bundling with Rust

```rust
let config = slint_build::CompilerConfiguration::new();
config.with_bundled_translations("path/to/translations");
```

The `<domain>` is the crate name.

### Bundling with C++

Set the `SLINT_BUNDLE_TRANSLATIONS` property in CMake:

```cmake
set_property(TARGET my_application PROPERTY SLINT_BUNDLE_TRANSLATIONS "${CMAKE_CURRENT_SOURCE_DIR}/lang")
```

> the `<domain>` is the cmake target name.

Reviewer: Similarly.... did you mean to use a more generic example path perhaps?

### Selecting a Translation

> If the `std` feature isenabled when building Slint, Slint will try to detect the language based on the locale.

Reviewer: ```suggestion
If you enable the `std` feature with Slint, language for translations is detected based on the locale.
```

If one of the bundled language matches the selected locale, it will be used.

> Use the `slint::select_bundled_translation` to change thanslation at runtime.

Reviewer: ```suggestion
Use the `slint::select_bundled_translation` function to change translations at runtime.
```
